CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -455,28 +600,353 @@ public static CVKImage Create(  CVKDevice cvkDevice,
+
+        // The PNG image format is RGB (24-bit)
+
+        // Size (in bytes) of the destination image including any padding
+        final long sourceImageSize = subResourceLayout.size(); 
+        // Offset (in bytes) that the image starts (usually 0)
+        long offset = subResourceLayout.offset();
+        // Calculate the file width including the padding (in bytes)
+        int fileWidth = (int)(subResourceLayout.rowPitch() / 4);
+        // Calculate the file height (in bytes)
+        int fileHeight = (int)(sourceImageSize / fileWidth) / 4;
+        // Calculate the file size (in bytes)
+        int fileSize = (int)(sourceImageSize / 4 ) * 3;
+
+        // TODO Hydra need to handle images with layers like the atlas texture
+        long fileLayers = subResourceLayout.depthPitch();
+
+        try {
+            // Create the output buffered image with the original images
+            // width and height, and in the PNG format (3 bytes BGR)
+            out = new BufferedImage(width, height, TYPE_3BYTE_BGR);
+            byte[] rgb = new byte[(int)fileSize];
+            int i = 0;
+            int r = 0;
+
+            ByteBuffer row = imageMemory.getByteBuffer((int)sourceImageSize);
+
+            for (int y = 0; y < fileHeight; y++)
+            {                   
+                for (int x = 0; x < fileWidth; x++)
+                {
+                    // Skip the padded bytes
+                    if (x >= width) {
+                        r += 4;
+                        continue;
+                    }
+                    // TODO: tests to see if we need to swizzle the rgb channels
+                    Byte red = row.get(r++);        // red
+                    Byte green = row.get(r++);      // green
+                    Byte blue = row.get(r++);       // blue
+                    Byte alpha = row.get(r++);      // alpha
+
+                    // If we need to swizzle colours manually (e.g. Blit is not supported which
+                    // does automatic conversion for us
+                    if (needsSwizzle) {
+                        rgb[i++] = blue;
+                        rgb[i++] = green;
+                        rgb[i++] = red;
+                    }
+                    else {
+                        rgb[i++] = red;
+                        rgb[i++] = green;
+                        rgb[i++] = blue;
+                    }
+                }
+            }
+
+            DataBuffer buffer = new DataBufferByte(rgb, rgb.length);
+
+            // 3 bytes per pixel: red, green, blue
+            WritableRaster raster = Raster.createInterleavedRaster(buffer, 
+                    width, height,          // width, height of image
+                    3 * width,              // number of bytes per row (stride)
+                    3,                      // number of channels (rgb)
+                    new int[] {0, 1, 2},    // colour format order (i.e. byte 0 is first, byte 1 is second ...
```

Comment:
Would this mean no swizzling above?

Reply:
Yes I think so, I need to test the swizzle case and see. Potentially if the copy/blit to an RGB format without Alpha works I can completely get rid of the loops traversing the pixels which would be great (and quicker!).